clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/core/AlwaysUpHealthIndicator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.core;
+
+import com.netflix.spectator.api.Registry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.atomic.AtomicReference;
+
+public abstract class AlwaysUpHealthIndicator implements HealthIndicator {
+
+  private static final Logger log = LoggerFactory.getLogger(AlwaysUpHealthIndicator.class);
+
+  protected final AtomicReference<Exception> lastException = new AtomicReference<>(null);
+  protected final AtomicReference<Boolean> hasInitialized = new AtomicReference<>(null);
+
+  private final AtomicLong errors;
+
+  public AlwaysUpHealthIndicator(Registry registry) {
+    this.errors = registry.gauge("health.amazon.errors", new AtomicLong(0));
```

Comment:
maybe stick a class tag on this gauge

Reply:
either that or the gauge name itself (probably not health.amazon at least)

Reply:
Oh, quite. A little too much copypasta in this place.